tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/NuGetPack.cs

```diff
@@ -0,0 +1,376 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using NuGet;
+using NuGet.Versioning;
+using NuGet.Packaging;
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Common;
+
+namespace Microsoft.DotNet.Build.Tasks.Packaging
+{
+    public class NuGetPack : PackagingTask
```

Comment:
There were a couple of issues I found that suggested NuGetPack was not planned to be supported or encouraged long-term: https://github.com/dotnet/arcade/issues/1033 and https://github.com/dotnet/arcade/issues/383.  I wanted to wait and see what was decided on here so we don't lose this later, especially since our bootstrap arcade still has to move in lockstep with arcade-services' and arcade's.